### PR TITLE
Use empty object status list when the StatusPolicyNone is used

### PR DESCRIPTION
**Pull request by Liujingfang1**
This reduces the request size when StatusPolicyNone is used

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func main() {
 	initCmd := initcmd.NewCmdInit(f, ioStreams)
 	updateHelp(names, initCmd)
 	loader := manifestreader.NewManifestLoader(f)
-	invFactory := inventory.ClusterClientFactory{}
+	invFactory := inventory.ClusterClientFactory{StatusPolicy: inventory.StatusPolicyAll}
```
Comment (karlkfi) [inline cmd/main.go:59]: do we need this if the default ConfigMap inventory doesn't store the status anyway? That's why i had left it to default to None before.

**Review by karlkfi — CHANGES_REQUESTED** [review]
The default inventory provider was deliberately StatusPolicyNone, because the custom inventory provider tests StatusPolicyAll. Why change it?

You might want to add some unit tests that test a custom provider so that we know the status update works when StatusPolicyAll and doesn't write them when StatusPolicyNone.

File: cmd/main.go (hunk 1 above)
Comment (Liujingfang1) [reply cmd/main.go]: We injected the apply status into the ConfigMap inventory. Updated the InventoryCM, now we don't need to change this to All.

**Comment by Liujingfang1** [comment]
> The default inventory provider was deliberately StatusPolicyNone, because the custom inventory provider tests StatusPolicyAll. Why change it?

We injected the apply status into the inventory configmap. If the passed in `[]ObjectStatus` is empty, it throws an error. Updated the code to allow this.

> You might want to add some unit tests that test a custom provider so that we know the status update works when StatusPolicyAll and doesn't write them when StatusPolicyNone.

Added an updated unit tests for this.

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**